pkg/logs/input/listener/tcp.go

```diff
@@ -111,6 +124,9 @@ func (l *TCPListener) startListener() error {
 
 // read reads data from connection, returns an error if it failed and stop the tailer.
 func (l *TCPListener) read(tailer *Tailer) ([]byte, error) {
+	if l.idleTimeout != 0 {
```

Comment:
NIT: Perhaps `l.idleTimeout > 0`? Since a negative number is technically valid input, but functionally undesirable.